apps/evm/src/components/DomainNameOrEllipseAddress/index.tsx

```diff
@@ -0,0 +1,77 @@
+import { useGetAddressDomainName } from 'clients/api';
+import { Icon, Tooltip } from 'components';
+import EllipseAddress, { type EllipseAddressProps } from 'components/EllipseAddress';
+import { useTranslation } from 'libs/translations';
+import { useChainId } from 'libs/wallet';
+import { ChainId } from 'types';
+import { cn, truncateAddress } from 'utilities';
+import type { Address } from 'viem';
+
+type DomainNameAddressProps = {
+  showProvider?: boolean;
+  showTooltip?: boolean;
+  shouldEllipseAddress?: boolean;
+} & EllipseAddressProps;
+
+export const DomainNameOrEllipseAddress: React.FC<DomainNameAddressProps> = ({
+  address,
+  className,
+  showProvider = true,
+  showTooltip = true,
+  shouldEllipseAddress = true,
+  ...ellipseAddressProps
+}) => {
+  const { t } = useTranslation();
+  const { chainId } = useChainId();
+  const { data: domainNames } = useGetAddressDomainName({
+    accountAddress: address as Address,
+  });
+
+  const chainDomainName = domainNames?.[chainId];
+  // Ethereum and Sepolia use ENS, other chains use SpaceID
+  const providerIcon =
+    chainId === ChainId.ETHEREUM || chainId === ChainId.SEPOLIA ? 'ensLogo' : 'spaceIdLogo';
+  const providerText =
+    chainId === ChainId.ETHEREUM || chainId === ChainId.SEPOLIA
+      ? t('web3DomainNames.providers.ens')
+      : t('web3DomainNames.providers.spaceId');
+
+  return (
+    <>
+      {chainDomainName ? (
+        <div className={cn('flex flex-row items-center', className)}>
+          {showProvider && (
```

Comment:
nit: this one is more of a trick than anything, but you can add the `space-x-1` class so that all the children are separated by 1 unit of spacing (and you can remove the `mr-` classes from the children).